Packages/com.chisel.editor/Chisel/Editor/Editor/MaterialBrowser/ChiselMaterialBrowserUtilities.cs

```diff
@@ -0,0 +1,242 @@
+/* * * * * * * * * * * * * * * * * * * * * *
+Chisel.Editors.ChiselMaterialBrowserCache.cs
+
+License: MIT (https://tldrlegal.com/license/mit-license)
+Author: Daniel Cornelius
+
+* * * * * * * * * * * * * * * * * * * * * */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Chisel.Components;
+using UnityEditor;
+using UnityEngine;
+
+namespace Chisel.Editors
+{
+    internal static class ChiselMaterialBrowserUtilities
+    {
+        // checks a path and returns true/false if a material is ignored or not
+        public static bool IsValidEntry( ChiselMaterialBrowserTile tile )
+        {
+            // these are here to clean things up a little bit and make it easier to read
+
+            bool PathContains( string path )
+            {
+                return tile.path.ToLower().Contains( path );
```

Comment:
Better to use Contains(path, StringComparison .InvariantCultureIgnoreCase) it avoids the garbage being created with ToLower

Reply:
Oh I guess unity's C# version doesn't have that, but you can still use IndexOf(String value, StringComparison comparisonType);

Reply:
This is now done, I just gotta push changes. I'll resolve this once that is done.